src/js/services/request.js

```diff
@@ -27,10 +27,9 @@ angular.module('app')
       // utils
       function setOptions(options) {
         options.headers = options.headers || {};
-        options.timeout = options.timeout || 5000;
+        options.timeout = options.timeout || 20000;
```

Comment:
Why default to to such a high timeout? 3000-5000 sounds like a good sweet-spot IMO.